scripts/apps/desks/views/desk-select.html

```diff
@@ -1,13 +1,16 @@
 <div class="dropdown dropdown--noarrow dropdown--dark dropdown--full-width" dropdown is-open="isOpen" tabindex="">
     <button class="dropdown__toggle dropdown__toggle--line" dropdown__toggle data-test-id="desk-select--handle">
         <span ng-show="!selectedDesk" translate>Select desk</span>
-        <span ng-show="selectedDesk">{{ defaultDesk && defaultDesk._id === selectedDesk._id ? 'Default' : selectedDesk.name | translate}}</span>
+        <span ng-show="selectedDesk">{{ defaultDesk && defaultDesk._id === selectedDesk._id ? 'Default' : selectedDesk.name || selectedDesk._id | translate}}</span>
         <b class="dropdown__caret white"></b>
     </button>
     <ul class="dropdown__menu" data-test-id="desk-select--options">
         <li ng-show="filter"><span translate>filter: "{{filter}}"</span></li>
         <li ng-repeat="desk in desks | orderBy:deskPreferenceOrdering track by desk._id">
             <button ng-click="onChange({desk: desk})" sd-desk-select-item>{{ :: desk.name }}</button>
         </li>
+        <li ng-if="sendToPersonal">
+            <button ng-click="onChange({desk: {'_id': 'Personal Space'}})">Personal Space</button>
```

Comment:
> If you are changing the component in a way that allows selecting something other than a desk, it would be good to update naming and structure too. For example, you could add type definitions to scope and instead desk property use
target: {deskId: string} | 'PERSONAL_SPACE
and then do
onChange('PERSONAL_SPACE')

I don't think this is the right solution because if I do this, I am getting an error `TypeError: Cannot use 'in' operator to search for 'selectDesk' in PERSONAL_SPACE` instead of this what I say - if we define the personal space like this in DeskSelect `scope.PERSONAL_SPACE = {label: gettext('Personal Space'), value: 'PERSONAL_SPACE'}` and then use it like this `onChange({desk: PERSONAL_SPACE})`

Reply:
> TypeError: Cannot use 'in' operator to search for 'selectDesk' in PERSONAL_SPACE

Not sure how and why are you trying to use `in` operator. You shouldn't need that. Here's an example:

```
interface IScope {
    target: {deskId: string} | 'PERSONAL_SPACE';
    setTarget(target: {deskId: string} | 'PERSONAL_SPACE'): void;
}

function link(scope: IScope) {
    scope.target = 'PERSONAL_SPACE';

    scope.setTarget = (t) => {
        scope.target = t;
    };

    // example
    scope.setTarget({deskId: 'abc'});
    scope.setTarget('PERSONAL_SPACE');
}
```

If it doesn't work you can go with the approach you suggested.